Taken from upstream.

From 7f549d02ac997505a7ed8a05339e73b851a875b8 Mon Sep 17 00:00:00 2001
From: Janneke Nieuwenhuizen <janneke@gnu.org>
Date: Sun, 9 Jun 2024 08:18:35 +0200
Subject: [PATCH 1/4] build: Remove a0-call-trunc-int from xfail for
 mescc-x86_64.

Reported by Vagrant Cascadian <vagrant@debian.org> via IRC.

* build-aux/check-mescc.sh (xfail_tests)[mescc x86_64]: Remove
a0-call-trunc-int.c.
---
 build-aux/check-mescc.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build-aux/check-mescc.sh b/build-aux/check-mescc.sh
index fb5e1ef46..b2d771c89 100755
--- a/build-aux/check-mescc.sh
+++ b/build-aux/check-mescc.sh
@@ -273,7 +273,6 @@ lib/tests/scaffold/91-goto-array.c
 lib/tests/scaffold/16-cast.c
 lib/tests/scaffold/17-compare-unsigned-le.c
 lib/tests/scaffold/17-compare-rotated.c
-lib/tests/scaffold/a0-call-trunc-int.c
 "
     fi
 
-- 
2.41.0

From 3389cf4aef17cec38a5fd693c4c5d62520b0eab3 Mon Sep 17 00:00:00 2001
From: Janneke Nieuwenhuizen <janneke@gnu.org>
Date: Sun, 9 Jun 2024 08:28:13 +0200
Subject: [PATCH 2/4] lib: arm: Fix signal.h header.

This is a follow-up to commit
    83a08a0788a5417d0a0bae3a43dc9c4026e09d99
    lib: Add ucontext_t for all architectures.

* include/linux/arm/signal.h: Add missing semicolon.
---
 include/linux/arm/signal.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/arm/signal.h b/include/linux/arm/signal.h
index 06cb09176..381811714 100644
--- a/include/linux/arm/signal.h
+++ b/include/linux/arm/signal.h
@@ -1,6 +1,7 @@
 /* -*-comment-start: "//";comment-end:""-*-
  * GNU Mes --- Maxwell Equations of Software
  * Copyright © 2024 Ekaitz Zarraga <ekaitz@elenq.tech>
+ * Copyright © 2024 Janneke Nieuwenhuizen <janneke@gnu.org>
  *
  * This file is part of GNU Mes.
  *
@@ -20,7 +21,7 @@
 
 // Taken from musl libc (4a16ddf5)
 
-typedef int greg_t
+typedef int greg_t;
 typedef int gregset_t[18];
 typedef struct sigcontext
 {
-- 
2.41.0

From 835d13ee2614a8b6ed54e5b7ead44b3d93cd63b4 Mon Sep 17 00:00:00 2001
From: Janneke Nieuwenhuizen <janneke@gnu.org>
Date: Sun, 9 Jun 2024 11:20:42 +0200
Subject: [PATCH 3/4] build: Add 90-signal to xfail for mescc-arm.

Bisected until 0.23; this test sadly never worked on arm.

* build-aux/check-mescc.sh (xfail_tests)[mescc-arm]: Add 90-signal.c
---
 build-aux/check-mescc.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build-aux/check-mescc.sh b/build-aux/check-mescc.sh
index b2d771c89..3900601e9 100755
--- a/build-aux/check-mescc.sh
+++ b/build-aux/check-mescc.sh
@@ -280,6 +280,7 @@ lib/tests/scaffold/17-compare-rotated.c
         xfail_tests="$xfail_tests
 lib/tests/scaffold/17-compare-rotated.c
 lib/tests/scaffold/67-m1-overflow-check.c
+lib/tests/signal/90-signal.c
 "
     fi
 
-- 
2.41.0

From 1683f074a0462bc27023ce5a980ff3d35d4126cf Mon Sep 17 00:00:00 2001
From: Janneke Nieuwenhuizen <janneke@gnu.org>
Date: Sun, 9 Jun 2024 11:24:25 +0200
Subject: [PATCH 4/4] build: Remove 17-compare-rotated, 67-m1-overflow-check
 xfail for mescc-arm.

These were fixed by the riscv64 work.

* build-aux/check-mescc.sh (xfail_tests)[mescc arm]: Remove
17-compare-rotated, 67-m1-overflow-check.
---
 build-aux/check-mescc.sh | 2 --
 1 file changed, 2 deletions(-)

diff --git a/build-aux/check-mescc.sh b/build-aux/check-mescc.sh
index 3900601e9..9abf43384 100755
--- a/build-aux/check-mescc.sh
+++ b/build-aux/check-mescc.sh
@@ -278,8 +278,6 @@ lib/tests/scaffold/17-compare-rotated.c
 
     if test $mes_cpu = arm; then
         xfail_tests="$xfail_tests
-lib/tests/scaffold/17-compare-rotated.c
-lib/tests/scaffold/67-m1-overflow-check.c
 lib/tests/signal/90-signal.c
 "
     fi
-- 
2.41.0

